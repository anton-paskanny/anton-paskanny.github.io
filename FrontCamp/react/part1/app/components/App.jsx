import React from 'react';

import BlogsCreationForm from './form/BlogsCreationForm.jsx';
import BlogsList from './blog/BlogsList.jsx';
import Filter from './filter/Filter.jsx';

import "./styles.css";

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      blogs: []
    };

    this.addBlog = this.addBlog.bind(this);
    this.filterByAuthor = this.filterByAuthor.bind(this);
  }
  addBlog(blog) {
    this.setState(prevState => ({
      blogs: [ ...prevState.blogs, blog]
    }));
  }
  filterByAuthor() {
    var filteredBlogs = this.state.blogs.sort((a, b) => {
      if (a.author < b.author) return -1;
      if (a.author > b.author) return 1;

      return 0;
    });

    this.setState({
      blogs: filteredBlogs
    });
  }
  render() {
    return (
      <React.Fragment>
        <h1>Simple blogs app</h1>
        <BlogsCreationForm addBlog={this.addBlog} />
        <BlogsList blogs={this.state.blogs} />
        { this.state.blogs.length > 1 && <Filter filterByAuthor={this.filterByAuthor} /> }
      </React.Fragment>
    )
  }
}
