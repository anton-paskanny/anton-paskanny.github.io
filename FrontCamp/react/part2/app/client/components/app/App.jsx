import React from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';

import AuthRoute from '../../containers/AuthRoute.jsx';
import Blogs from '../../containers/Blogs.jsx';
import Header from '../../containers/Header.jsx';
import Intro from '../intro/Intro.jsx';
import SignUp from '../../containers/SignUp.jsx';
import SignIn from '../../containers/SignIn.jsx';

import './styles.css';
import "./normalize.css";

export default class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <React.Fragment>
        <Header />
        <div className="app__wrapper">
          <Switch>
            <Route exact path="/" component={Intro} />
            <Route path="/signup" component={SignUp} />
            <Route path="/signin" component={SignIn} />
            <AuthRoute path="/blogs" component={<Blogs />} />
          </Switch>
        </div>
      </React.Fragment>
    )
  }
}
