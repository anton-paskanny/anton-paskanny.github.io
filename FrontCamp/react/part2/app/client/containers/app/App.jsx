import React from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import { connect } from 'react-redux';

import AuthRoute from '../authRoute/AuthRoute.jsx';
import Blogs from '../blogs/Blogs.jsx';
import Header from '../header/Header.jsx';
import Intro from '../../components/intro/Intro.jsx';
import SignUp from '../forms/auth/signup/SignUp.jsx';
import SignIn from '../forms/auth/signin/SignIn.jsx';

import './styles.css';
import "./normalize.css";


export default class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <React.Fragment>
        <Header />
        <div className="app__wrapper">
          <Switch>
            <Route exact path="/" component={Intro} />
            <Route path="/signup" component={SignUp} />
            <Route path="/signin" component={SignIn} />
            <AuthRoute path="/blogs" component={<Blogs />} />
          </Switch>
        </div>
      </React.Fragment>
    )
  }
}
