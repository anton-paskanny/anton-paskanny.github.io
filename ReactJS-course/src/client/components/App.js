import React from 'react';
import { Route, Switch, BrowserRouter as Router } from 'react-router-dom';

import Header from './Header/Header';
import HeaderDetail from './Header/HeaderDetail';
import MovieResults from '../containers/MovieResults';
import Footer from './Footer/Footer';

const App = () => (
    <Router>
        <div className="app-wrapper">
            <Switch>
                <Route path="/" component={Header} exact/>
                <Route path="/movie/:id" component={HeaderDetail} />
            </Switch>
            <MovieResults />
            <Footer />
        </div>
    </Router>
);

export default App;