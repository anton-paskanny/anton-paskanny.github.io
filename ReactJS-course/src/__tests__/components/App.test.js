import React from 'React';
import renderer from 'react-test-renderer';

import App from '../../client/components/App';

test('Render correctly App component', () => {
    const component = renderer.create(<App />).toJSON();
    expect(component).toMatchSnapshot();
});

