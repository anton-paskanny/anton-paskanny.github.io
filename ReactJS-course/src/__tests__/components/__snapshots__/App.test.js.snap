// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly App component 1`] = `
<div
  className="app-wrapper"
>
  <header>
    <h3>
      netflixroulette
    </h3>
    <div
      className="search"
    >
      <h4>
        Find your movie
      </h4>
      <div
        className="search__input-wrapper"
      >
        <input
          name="movie"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="search-by"
      >
        <p>
          Search by
        </p>
        <ul>
          <li>
            <button>
              Title
            </button>
          </li>
          <li>
            <button>
              Genre
            </button>
          </li>
        </ul>
        <button>
          Search
        </button>
      </div>
    </div>
  </header>
  <div
    className="results"
  >
    <div>
      Sort Panel
    </div>
    <div
      className="results__items"
    >
      No items were found
    </div>
  </div>
  <footer
    className="footer"
  >
    <h3>
      netflixroulette
    </h3>
  </footer>
</div>
`;
